mailbox/store/src/test/java/org/apache/james/mailbox/store/mail/model/AbstractAttachmentMapperTest.java

```diff
@@ -60,46 +65,113 @@ public void tearDown() throws MailboxException {
     }
 
     @Test
-    public void addShouldWork() throws IOException {
+    public void putShouldWork() throws IOException {
         // Given
-        String blobId = "blobId";
-        String expected = "content";
+        InMemoryAttachmentId blobId = InMemoryAttachmentId.of(1);
+        String expectedContent = "content";
+        Content content = new ByteContent(expectedContent.getBytes(Charsets.UTF_8));
+        Attachment newAttachment = new SimpleAttachment(blobId, content);
 
         // When
-        attachmentMapper.add(blobId, IOUtils.toInputStream(expected, Charsets.UTF_8));
+        attachmentMapper.put(newAttachment);
 
         // Then
-        InputStream attachment = attachmentMapper.get(blobId);
-        assertThat(IOUtils.toString(attachment, Charsets.UTF_8)).isEqualTo(expected);
+        Attachment actual = attachmentMapper.get(blobId);
+        String actualContentAsString = IOUtils.toString(actual.getContent().getInputStream(), Charsets.UTF_8);
+        assertThat(actualContentAsString).isEqualTo(expectedContent);
     }
 
     @Test
     public void getShouldWork() throws IOException {
         // Given
-        String blobId = "blobId";
-        String expected = "content";
-
-        attachmentMapper.add(blobId, IOUtils.toInputStream(expected, Charsets.UTF_8));
+        String expectedContent = "content";
+        InMemoryAttachmentId blobId = InMemoryAttachmentId.of(1);
+        saveAsNewAttachment(blobId, expectedContent);
 
         // When
-        InputStream attachment = attachmentMapper.get(blobId);
+        Attachment actual = attachmentMapper.get(blobId);
+        String actualContentAsString = IOUtils.toString(actual.getContent().getInputStream(), Charsets.UTF_8);
 
         // Then
-        assertThat(IOUtils.toString(attachment, Charsets.UTF_8)).isEqualTo(expected);
+        assertThat(actualContentAsString).isEqualTo(expectedContent);
     }
 
     @Test
     public void deleteShouldWork() throws IOException {
         // Given
-        String blobId = "blobId";
-        String expected = "content";
-
-        attachmentMapper.add(blobId, IOUtils.toInputStream(expected, Charsets.UTF_8));
+        String expectedContent = "content";
+        final InMemoryAttachmentId blobId = InMemoryAttachmentId.of(1);
+        saveAsNewAttachment(blobId, expectedContent);
 
         // When
-        attachmentMapper.remove(blobId);
+        attachmentMapper.delete(blobId);
 
         // Then
-        assertThat(attachmentMapper.get(blobId)).isNull();
+        try {
+            attachmentMapper.get(blobId);
+            fail("Should have thrown while accessing removed item !");
```

Comment:
using java 6 api only and assertj <= 1.* ?
If you have suggestions, feel free to provide an example, please.